src/averageAges.js

```diff
@@ -55,7 +81,26 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const womenWithChild = people.filter(
+    (woman) =>
+      woman.sex === 'f' && people.some(
+        (child) => child.mother === woman.name)
+  );
+  const children = people.filter(
+    (child) =>
+      womenWithChild.some(
+        (mother) => child.mother === mother.name)
+      && (onlyWithSon ? child.sex === 'm' : true)
+  );
```

Comment:
```suggestion
  const children = people.filter(child =>
    onlyWithSon
      ? people.some(mother => child.mother === mother.name) && child.sex === 'm'
      : people.some(mother => child.mother === mother.name)
  );
```
І це можна було б скоротити